newrelic/resource_newrelic_nrql_alert_condition.go

```diff
@@ -349,6 +349,10 @@ func resourceNewRelicNrqlAlertCondition() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"CADENCE", "EVENT_FLOW", "EVENT_TIMER"}, true),
 				Description:  "The method that determines when we consider an aggregation window to be complete so that we can evaluate the signal for violations. Default is CADENCE.",
 				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					sinceValueExists := d.Get("nrql.0.since_value") != nil
+					if sinceValueExists {
+						return false
+					}
```

Comment:
:memo: We will not suppress the diff the user is using a deprecated `since_value` field in their nrql alert condition resource. 

This is useful if the user has an existing alert condition made from terraform, then makes irreversible changes in the UI, and wants to revert back to the original alert condition made from terraform. 

If we do not have this check, the user the will be unable to recover their original condition and be forced to migrate because of changes made in the UI.